data/processing.py

```diff
@@ -221,138 +199,134 @@ def load_data():
-        row = https_row_for(domain)
 
-        # Needs to be enabled, with issues is allowed
-        if row[LABELS['https']] >= 1:
-          https_stats['https'] += 1
+    # HTTPS
+    eligible = Domain.eligible_for_agency(agency['slug'], 'https')
 
-        # Needs to be Default or Strict to be 'Yes'
-        if row[LABELS['https_forced']] >= 2:
-          https_stats['https_forced'] += 1
+    agency_report = {
+      'eligible': len(eligible),
+      'uses': 0,
+      'enforces': 0,
+      'hsts': 0,
+      'grade': 0
+    }
 
-        # Needs to be at least partially present
-        if row[LABELS['hsts']] >= 1:
-          https_stats['hsts'] += 1
+    for domain in eligible:
+      report = domain['https']
 
-        # Needs to be A- or above
-        if row[LABELS['grade']] >= 4:
-          https_stats['grade'] += 1
+      # Needs to be enabled, with issues is allowed
+      if report['uses'] >= 1:
+        agency_report['uses'] += 1
 
-      if evaluating_for_analytics(domain):
+      # Needs to be Default or Strict to be 'Yes'
+      if report['enforces'] >= 2:
+        agency_report['enforces'] += 1
 
-        analytics_total += 1
-        row = analytics_row_for(domain)
+      # Needs to be at least partially present
+      if report['hsts'] >= 1:
+        agency_report['hsts'] += 1
 
-        # Enabled ('Yes')
-        if row[LABELS['dap']] >= 1:
-          analytics_stats['dap'] += 1
+      # Needs to be A- or above
+      if report['grade'] >= 4:
+        agency_report['grade'] += 1
 
+    print("[%s][%s] Adding report." % (agency['slug'], 'https'))
+    Agency.add_report(agency['slug'], 'https', agency_report)
 
-    if https_total > 0:
-      row = {
-        'Agency': agency,
-        'Number of Domains': https_total,
-        LABELS['https']: percent(https_stats['https'], https_total),
-        LABELS['https_forced']: percent(https_stats['https_forced'], https_total),
-        LABELS['hsts']: percent(https_stats['hsts'], https_total),
-        LABELS['grade_agencies']: percent(https_stats['grade'], https_total)
-      }
-      https_agencies.append(row)
-      agency_map[slugify.slugify(agency)]['https'] = row
-    else:
-      agency_map[slugify.slugify(agency)]['https'] = None
-
-    if analytics_total > 0:
-      row = {
-        'Agency': agency,
-        'Number of Domains': analytics_total,
-        LABELS['dap']: percent(analytics_stats['dap'], analytics_total)
-      }
-      analytics_agencies.append(row)
-      agency_map[slugify.slugify(agency)]['analytics'] = row
-    else:
-      agency_map[slugify.slugify(agency)]['analytics'] = None
 
+def eligible_for_https(domain):
+  return (domain["live"] == True)
 
-def evaluating_for_https(domain):
+def eligible_for_analytics(domain):
   return (
-    (domain_data[domain].get('inspect') is not None) and
-    (domain_data[domain]['inspect']["Live"] == "True")
+    (domain["live"] == True) and
+    (domain["redirect"] == False) and
+    (domain["branch"] == "executive")
```

Comment:
It's expected that any field needed to calculate "eligibility" for any particular report type (including any future ones) will be put on the top-level of the `Domain` object, even if that means it's data that was brought in from `site-inspector` and `inspect.csv`, rather than the domain file itself.